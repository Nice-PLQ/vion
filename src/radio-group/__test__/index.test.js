import { mount } from '@vue/test-utils';
import VRadioGroup from '../radio-group.vue';
import VRadio from '../../radio/radio.vue';

describe('RadioGroup', () => {
  test('单选框组渲染', async () => {
    const wrapper = mount({
      template: `
        <v-radio-group>
          <v-radio value="a">单选A</v-radio>
          <v-radio value="b">单选B</v-radio>
        </v-radio-group>
      `
    }, {
      global: {
        components: { VRadioGroup, VRadio }
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('单选', async () => {
    const wrapper = mount({
      template: `
          <v-radio-group v-model="checked" vertical>
            <v-radio value="a">单选A</v-radio>
            <v-radio value="b">单选B</v-radio>
            <v-radio value="c">单选C</v-radio>
          </v-radio-group>
        `,
      data() {
        return {
          checked: ''
        };
      }
    }, {
      global: {
        components: { VRadioGroup, VRadio }
      }
    });
    const radio = wrapper.findAll('.v-radio');
    await radio[0].trigger('click');
    expect(wrapper.vm.checked).toBe('a');
    await radio[1].trigger('click');
    expect(wrapper.vm.checked).toBe('b');

    expect(wrapper.classes()).toContain('v-radio-group_vertical');
  });

  test('禁用', async () => {
    const wrapper = mount({
      template: `
          <v-radio-group v-model="checked" disabled>
            <v-radio value="a">单选A</v-radio>
            <v-radio value="b">单选B</v-radio>
            <v-radio value="c">单选C</v-radio>
          </v-radio-group>
        `,
      data() {
        return {
          checked: ''
        };
      }
    }, {
      global: {
        components: { VRadioGroup, VRadio }
      }
    });
    const radio = wrapper.findAll('.v-radio');
    await radio[1].trigger('click');
    expect(wrapper.vm.checked).toBe('');
  });
});
