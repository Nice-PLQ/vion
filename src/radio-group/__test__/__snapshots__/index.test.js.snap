// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup 单选框组渲染 1`] = `
<div
  class="v-radio-group"
>
  
  <div
    class="v-radio"
  >
    <div
      class="v-radio__icon"
    >
      <input
        class="v-radio__input"
        type="radio"
      />
      <!--v-if-->
    </div>
    <span
      class="v-radio__label"
    >
      
      单选A
      
    </span>
  </div>
  <div
    class="v-radio"
  >
    <div
      class="v-radio__icon"
    >
      <input
        class="v-radio__input"
        type="radio"
      />
      <!--v-if-->
    </div>
    <span
      class="v-radio__label"
    >
      
      单选B
      
    </span>
  </div>
  
</div>
`;
